Adds an insecure channel option

antonpperez opened the pull request:
- The option is available when creating the HandlerClient
  - Tests for the handler are missing for the new option as it would
  need some kind of different configuration for the docker image (since
  it is an integration test)
  - Adds test for the discovery client

krimvp committed:
Adds an insecure channel option

  - The option is available when creating the HandlerClient
  - Tests for the handler are missing for the new option as it would
  need some kind of different configuration for the docker image (since
  it is an integration test)
  - Adds test for the discovery client

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/TheThingsNetwork/python-app-sdk?pullRequest=33) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/TheThingsNetwork/python-app-sdk?pullRequest=33) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/TheThingsNetwork/python-app-sdk?pullRequest=33) it.</sub>

krimvp committed:
Add testing library to patch dependencies in unittests

  - Since there needs to be a support for python 2.7, 3.5 and 3.6 the
  default unittest.mock will not work. The dependency "mock" makes it
  work but it forced to rename the already existing mock.py (so it can
  be properly imported)

johanstokking commented:
Thanks @antonpperez, I'll get back asap

krimvp commented:
@johanstokking do you have an estimated ETA for this pull request review? We would like to integrate it into our CI/CD pipeline and would be better to pull the repo directly from the python artifactory rather than a Github branch

johanstokking commented:
@krimvp thanks! This is published as version `2.1.5`
